File: codeforces/1294-A.py
t = int(input())
for _ in range(t):
    *a, n = map(int, input().split())
    a.sort()
    n -= 2 * a[2] - a[1] - a[0]
    if n < 0 or n % 3 != 0:
        print("NO")
    else:
        print("YES")

File: codeforces/1294-A.cpp
#include <bits/stdc++.h>


using namespace std;
int main(){
    int t, a[3], n;
    cin >> t;
    while(t--){
        cin >> a[0] >> a[1] >> a[2] >> n;
        sort(a, a + 3);
        n -= 2 * a[2] - a[1] - a[0];
        if (n < 0 || n % 3 != 0){
            cout << "NO" << endl;
        }else{
            cout << "YES" << endl;
        }
    }
}